spyder/utils/programs.py

```diff
@@ -474,6 +474,24 @@ def is_python_interpreter(filename):
     if (not osp.isfile(real_filename) or encoding.is_text_file(real_filename)
         or not is_python_interpreter_valid_name(filename)):
         return False
+    elif (not encoding.is_text_file(real_filename)

```

Comment:
Sorry @dalthviz, you were right last time. The previous `if` checks if users are passing a real file, if it's binary and a valid interpreter name. So this line needs to be

    elif is_pythonw(filename):

and nothing else.